src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/EffectivePolicyMutationStatements.scala

```diff
@@ -0,0 +1,363 @@
+    val ep = EffectivePolicyTable.syntax("ep")
+    val epr = EffectivePolicyRoleTable.syntax(("epr"))
+    samsql"""delete from ${EffectivePolicyRoleTable as epr}
+              using ${EffectivePolicyTable as ep}
+              where ${ep.sourcePolicyId} = $policyPK
+              and ${epr.resourcePolicyId} = ${ep.id}""".update().apply()
+  }
+
+  /**
+    * Delete all actions on effecitve policies with source policy of policyPK.
+    * @param policyPK
+    * @param session
+    * @return
+    */
+  protected def deleteEffectivePolicyActions(policyPK: PolicyPK)(implicit session: DBSession): Int = {
+    val ep = EffectivePolicyTable.syntax("ep")
+    val epa = EffectivePolicyActionTable.syntax(("epa"))
+    samsql"""delete from ${EffectivePolicyActionTable as epa}
+              using ${EffectivePolicyTable as ep}
+              where ${ep.sourcePolicyId} = $policyPK
+              and ${epa.resourcePolicyId} = ${ep.id}""".update().apply()
+  }
+
+  protected def setEffectivePoliciesPublic(policyPK: PolicyPK, isPublic: Boolean)(implicit session: DBSession): Int = {
+    val ep = EffectivePolicyTable.syntax("ep")
+    val effectivePolicyTableColumn = EffectivePolicyTable.column
+    samsql"""update ${EffectivePolicyTable as ep}
+              set ${effectivePolicyTableColumn.public} = ${isPublic}
+              where ${ep.sourcePolicyId} = ${policyPK}""".update().apply()
+  }
+
+  /**
+    *
+    * Determining whether a role should or should not apply to a resource is a bit more complicated than it initially
+    * appears. This logic is shared across queries that search a resource's hierarchy for all of the relevant roles and actions
+    * that a user has on said resource. The following truth table shows the desired behavior of this SQL fragment where
+    * result indicates whether a given role does or does not apply to the resource.
+    * inherited is determined by sourcePolicy.resourceId != effectivePolicy.resourceId
+    *
+    *         inherited             |  policyRole.descendantsOnly  |  flattenedRole.descendantsOnly  |  result
+    *            T                  |             T                |                T                |    T
+    *            T                  |             T                |                F                |    T
+    *            T                  |             F                |                T                |    T
+    *            T                  |             F                |                F                |    F
+    *            F                  |             T                |                T                |    F
+    *            F                  |             T                |                F                |    F
+    *            F                  |             F                |                T                |    F
+    *            F                  |             F                |                F                |    T
+    *
```

Comment:
@marctalbott figured this out, if there were XOR in SQL it might be better but there isn't :(

Reply:
Yeah, the query looks good, but I was wondering about putting it above the truth table in the comment, just to make it more clear at a first glance. Although since we can't represent the sql that way, I'm okay with leaving it for now.